webhonc/src/index.ts

```diff
@@ -0,0 +1,94 @@
+import { zValidator } from "@hono/zod-validator";
+import { Hono } from "hono";
+import { z } from "zod";
+import type { Bindings, Variables } from "./types";
+import { WebHonc } from "./webhonc";
+
+const app = new Hono<{ Bindings: Bindings; Variables: Variables }>();
+
+app.get("/ws", async (c) => {
+  if (c.req.header("upgrade") !== "websocket") {
+    return new Response("Not a websocket request", { status: 426 });
+  }
+
+  const id = c.env.WEBHONC.newUniqueId();
+  const webhonc = c.env.WEBHONC.get(id) as DurableObjectStub<WebHonc>;
+
+  return webhonc.fetch(c.req.raw);
+});
+
+app.all(
+  "/:id/*",
+  zValidator("param", z.object({ id: z.string() })),
+  async (c) => {
+    const { id } = c.req.valid("param");
+
+    // We need to serialize the path components without the id
+    // so we can send it down the wire and replay it locally
+    const pathComponentsWithoutId = c.req.path
```

Comment:
still might prefer something that just replaces `/^\/id/`, feels more straightforward

Reply:
tbh I experimented a little bit here and found even the simplest regex approach a bit of a pain to read. I agree that sending path components as array items might be overkill so I can just join them back into a string before sending down?

Reply:
joining them could be good but let's do that later, eager to merge this